registry/storage/driver/gcs/gcs.go

```diff
@@ -689,42 +676,52 @@ func (d *driver) List(ctx context.Context, path string) ([]string, error) {
 // Move moves an object stored at sourcePath to destPath, removing the
 // original object.
 func (d *driver) Move(ctx context.Context, sourcePath string, destPath string) error {
-	_, err := storageCopyObject(ctx, d.bucket, d.pathToKey(sourcePath), d.bucket, d.pathToKey(destPath), d.gcs)
+	srcKey, dstKey := d.pathToKey(sourcePath), d.pathToKey(destPath)
+	src := d.bucket.Object(srcKey)
+	_, err := d.bucket.Object(dstKey).CopierFrom(src).Run(ctx)
 	if err != nil {
-		if status, ok := err.(*googleapi.Error); ok {
+		var status *googleapi.Error
+		if errors.As(err, &status) {
 			if status.Code == http.StatusNotFound {
-				return storagedriver.PathNotFoundError{Path: sourcePath}
+				return storagedriver.PathNotFoundError{Path: srcKey}
 			}
 		}
-		return err
+		return fmt.Errorf("move %q to %q: %w", srcKey, dstKey, err)
 	}
-	err = storageDeleteObject(ctx, d.bucket, d.pathToKey(sourcePath), d.gcs)
+	err = src.Delete(ctx)
 	// if deleting the file fails, log the error, but do not fail; the file was successfully copied,
 	// and the original should eventually be cleaned when purging the uploads folder.
 	if err != nil {
-		logrus.Infof("error deleting file: %v due to %v", sourcePath, err)
+		logrus.Infof("error deleting %v: %v", sourcePath, err)
 	}
 	return nil
 }
 
 // listAll recursively lists all names of objects stored at "prefix" and its subpaths.
 func (d *driver) listAll(ctx context.Context, prefix string) ([]string, error) {
-	list := make([]string, 0, 64)
-	query := &storage.Query{}
-	query.Prefix = prefix
-	query.Versions = false
-	objects, err := storageListObjects(ctx, d.bucket, query, d.gcs)
-	if err != nil {
-		return nil, err
+	query := &storage.Query{
+		Prefix:   prefix,
+		Versions: false,
 	}
-	for _, obj := range objects {
+	objects := d.bucket.Objects(ctx, query)
```

Comment:
Nit; as we're updating this code; looks like `query` is only used here, so might as well inline it;

```go
objects := d.bucket.Objects(ctx, &storage.Query{
	Prefix:   prefix,
	Versions: false,
})
```

Reply:
done: https://github.com/distribution/distribution/pull/4120/commits/12fbda6c6253e7ec79f373393d9c9b3290c216f2